datafusion/physical-expr/src/window/nth_value.rs

```diff
@@ -162,16 +165,27 @@ impl PartitionEvaluator for NthValueEvaluator {
             NthValueKind::Nth(n) => {
                 let n_range =
                     state.window_frame_range.end - state.window_frame_range.start;
-                (n_range >= (n as usize) && size >= (n as usize), false)
+                #[allow(clippy::comparison_chain)]
```

Comment:
Is there any particular reason not to follow clippy style (with cmp and match)? 

https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/comparison_chain.rs

Reply:
That Clippy lint is somewhat controversial and ignored by some people (see relevant GH issues). I actually agree that it is a bad lint and reduces readability, that's why I surgically disabled it while reviewing.

Not important enough to revert in a follow-on, but I wanted to explain the context around it. We haven't had a chance to sync with @mustafasrepo about it as I am traveling, so he was unaware why.

Reply:
Yeah, in general I find some of Clippy's requirements to reduce code readability compared to my preferred style. However, I have found just doing what clippy says does result in a more consistent codebase so I often just do it for the sake of not having to explain why I disabled clippy.